### ISSUE TITLE
coco object detection accuracy bugs

### ISSUE THREAD

**Issue by xhzhao** [tensorflow/models#6421]
i'm trying to repeat the pretrained object detection model accuracy on coco, and we could not repeat this mAP numbers after some suffering hacks. here is my result:
![image](https://user-images.githubusercontent.com/17486215/54798058-c294a080-4c91-11e9-9253-d9620d210c7b.png)

model | coco image   count | image resize | mAP
-- | -- | -- | --
ssd_resnet_50_fpn_coco | standard: 40504 | None | 37.97
ssd_resnet_50_fpn_coco | standard: 40504 | 640 | 36.97
ssd_resnet_50_fpn_coco | subset: 8059 | 640 | 31.67

Am i missing something?
This is the first time that the model zoo accuracy could not be repeated, and i thought tensorflow, as a leading framework, should not have this kind of navie bugs.

Please comment.

**Comment by xhzhao** [comment]
BTW, this config information is very miss leading: [iou_threshold](https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config#L129)

i checked the coco evaluation code, and this threshold is of no use:
https://github.com/tensorflow/models/blob/master/research/object_detection/metrics/coco_evaluation.py#L165

**Comment by xhzhao** [comment]
a related issue: https://github.com/tensorflow/models/issues/6278
the mAP from this issue is very close to mine: 30.97, and my result:

i bus id: 0000:09:00.0, compute capability: 6.1)
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8059/8059 [11:19<00:00, 12.99it/s]
Loading and preparing results...
DONE (t=0.11s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=24.92s).
Accumulating evaluation results...
DONE (t=3.19s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.317
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.473
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.349
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.120
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.339
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.479
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.271
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.389
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.397
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.143
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.424
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.599
{
    "avg_latency_ms": 62.08372474056954,
    "avg_throughput_fps": 16.10728100124017,
    "map": 0.31666184451309204
}

**Comment by rootkitchao** [comment]
I think iou_threshold has nothing to do with evaluation, this is part of the model.Have you created a validation set based on mscoco_minival_ids.txt?

**Comment by xhzhao** [comment]
@rootkitchao yes, we extract the standard annotation file(instances_val2014.json) to a subset(instances_val2014_8k.json) with this ids.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
